pkg/auth/authproviders/oidc/backend_impl.go

```diff
@@ -630,9 +649,66 @@ func userInfoToExternalClaims(userInfo *userInfoType) *tokens.ExternalUserClaim
 	return claim
 }
 
-func (p *backendImpl) injectHTTPClient(ctx context.Context) context.Context {
-	if p.httpClient != nil {
-		return oidc.ClientContext(ctx, p.httpClient)
+type claimExtractor interface {
+	Claims(v interface{}) error
+}
+
+func extractCustomClaims(externalUserClaim *tokens.ExternalUserClaim, mappings map[string]string, claimExtractor claimExtractor) error {
+	claims := make(map[string]interface{}, 0)
+	if err := claimExtractor.Claims(&claims); err != nil {
+		return errors.Wrap(err, "failed to extract claims")
 	}
-	return ctx
+	for fromClaimPath, toClaimName := range mappings {
+		val, err := extractClaimFromPath(fromClaimPath, claims)
+		if err != nil {
+			log.Debugf("Failed to extract claim from path: %v", err)
+			continue
+		}
+		if err := addClaimToUserClaims(externalUserClaim, toClaimName, val); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func addClaimToUserClaims(externalUserClaim *tokens.ExternalUserClaim, attributeName string, claimValue interface{}) error {
+	switch v := claimValue.(type) {
+	case []interface{}:
+		for i, arrayVal := range v {
+			_, isArray := arrayVal.([]interface{})
+			_, isNestedStruct := arrayVal.(map[string]interface{})
+			if isArray || isNestedStruct {
+				return errors.Errorf("Unsupported claim type %T with value %+v", arrayVal, arrayVal)
```

Comment:
```suggestion
				return errors.Errorf("unsupported claim type %T with value %+v", arrayVal, arrayVal)
```
We strive to standardize on lower case error messages.